### ISSUE TITLE
Upload and Embed HTML5 Media Types

### ISSUE THREAD

**Issue by Praveen-Rai** [quilljs/quill#2232]
I would like to suggest to have inbuilt support to upload and embed media types from local machine. For e.g. this plugin for image upload https://github.com/fxmontigny/quill-image-upload.

The idea is to have support for all the basic media types supported by HTML5, namely Video, Images, Audio. For Video uploads quill must support multiple source selection and generate the HTML with those sources.

**Comment by DmitrySkripkin** [comment]
Upload process can be completely different in different projects. I don't think we can make universal module for upload.

**Comment by Praveen-Rai** [comment]
I can understand that, but atleast the front-end functionalities. But, there's something common in the front-end. For e.g. if i wish to add an image in the Quill-Editor, i wish to be able to browse the file, call some api/route at back-end to upload file (this can be part of Quill configuration) and replace it with returned path ( we can suggest the developers to return the path from the api/route ). 

While the real uploading part is actually scenario driven, the other two has a lot to do on front-end and can be generalized. There are lot of use cases, and let me take a few of them 

 **Attaching a Video** - If the html has to display the video correctly and not just a hyperlink, it has to be encapsulated in the HTML Video tag. And as you understand that multiple sources are to be provided for the HTML Video Tag, which add a requirement to allow multiple file uploads. 
**Upload Progress** - Everyone want to display the uploaded files and their progress. 
**Filtering** - When uploading images, the user must be allowed to select the image formats, and so on.

Basically, what i am trying to suggest is a basic framework and front-end functionality, even if we make it very specific to Quill. And we can still let the developers to override or add their custom functionalities using configuration files.

Does it make sense ?
